kubepy/appliers.py

```diff
@@ -174,8 +177,26 @@ def raise_if_failed(self):
                 self._raise_for_failed_condition(condition)
         failures = self.status.get('failed', 0)
         if self.max_retries and failures > self.max_retries:
+            failed_pod = self.get_job_pod()
+            if failed_pod:
+                stdout, stderr = self.get_failed_pod_logs(failed_pod)
+            else:
+                stdout, stderr = '', ''
+            for line in str(stdout).split('\\n'):
+                print('STDOUT:\t', line)
+            for line in str(stderr).split('\\n'):
+                print('STDERR:\t', line)
             raise JobError('Job failed {} times.'.format(failures))
 
+    def get_failed_pod_logs(self, failed_pod):
+        for init_container_status in failed_pod['status'].get('initContainerStatuses', []):
+            if init_container_status['state'].get('terminated', {}).get('reason') == 'ContainerCannotRun':
+                return 'Init container failed!', init_container_status['state']['terminated']['message']
```

Comment:
It's misleading to return an arbitrary string and pretend it came from stdout.
As mentioned above: maybe let's drop stdout/stderr distinction?